modules/nextflow/src/main/groovy/nextflow/container/ContainerConfig.groovy

```diff
@@ -17,129 +17,51 @@
 package nextflow.container
 
 import groovy.transform.CompileStatic
-import groovy.transform.Memoized
-import groovy.util.logging.Slf4j
 
 /**
- * Models container engine configuration
+ * Models generic container configuration
  *
  * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
  */
-@Slf4j
 @CompileStatic
-class ContainerConfig extends LinkedHashMap {
+interface ContainerConfig {
 
-    private Map<String,String> sysEnv = System.getenv()
-
-    /* required by Kryo deserialization -- do not remove */
-    private ContainerConfig() { }
-
-    ContainerConfig(Map config) {
-        super(config)
-    }
-
-    ContainerConfig(Map config, Map<String,String> env) {
-        super(config)
-        this.sysEnv = env
+    default boolean canRunOciImage() {
+        return false
     }
 
-    boolean isEnabled() {
-        return get('enabled')?.toString() == 'true'
+    default boolean entrypointOverride() {
+        return ContainerHelper.entrypointOverride()
     }
```

Comment:
The `entrypointOverride` is a hidden config option that can be set for `docker`, `k8s`, and the other container config scopes.

I made it so that it can only be enabled by setting the `NXF_CONTAINER_ENTRYPOINT_OVERRIDE` environment variable, since that seems simpler than adding a config option for every container scope.

If we want to preserve this behavior, I suggest we document this env var and add a note to the 25.10 migration notes.

Reply:
Related: https://github.com/nextflow-io/nextflow/commit/b3a4bf857cef0a8b16814baf9e13aa9296ca208a